File: internal/content/testdata/bench/go/nc-combination-sum.go
import "sort"

func combination_sum(candidates []int, target int) [][]int {
	sort.Ints(candidates)
	res := [][]int{}
	var backtrack func(start, remaining int, current []int)
	backtrack = func(start, remaining int, current []int) {
		if remaining == 0 {
			tmp := make([]int, len(current))
			copy(tmp, current)
			res = append(res, tmp)
			return
		}
		for i := start; i < len(candidates); i++ {
			if candidates[i] > remaining {
				break
			}
			current = append(current, candidates[i])
			backtrack(i, remaining-candidates[i], current)
			current = current[:len(current)-1]
		}
	}
	backtrack(0, target, []int{})
	for _, c := range res {
		sort.Ints(c)
	}
	sort.Slice(res, func(i, j int) bool {
		a, b := res[i], res[j]
		minLen := len(a)
		if len(b) < minLen {
			minLen = len(b)
		}
		for k := 0; k < minLen; k++ {
			if a[k] != b[k] {
				return a[k] < b[k]
			}
		}
		return len(a) < len(b)
	})
	return res
}

File: internal/content/testdata/bench/rust/nc-combination-sum.rs
fn backtrack(
    start: usize,
    candidates: &Vec<i64>,
    current: &mut Vec<i64>,
    remaining: i64,
    res: &mut Vec<Vec<i64>>,
) {
    if remaining == 0 {
        res.push(current.clone());
        return;
    }
    for i in start..candidates.len() {
        if candidates[i] > remaining {
            break;
        }
        current.push(candidates[i]);
        backtrack(i, candidates, current, remaining - candidates[i], res);
        current.pop();
    }
}

fn combination_sum(candidates: Vec<i64>, target: i64) -> Vec<Vec<i64>> {
    let mut candidates = candidates;
    candidates.sort();
    let mut res: Vec<Vec<i64>> = Vec::new();
    let mut current: Vec<i64> = Vec::new();
    backtrack(0, &candidates, &mut current, target, &mut res);
    for c in res.iter_mut() {
        c.sort();
    }
    res.sort();
    res
}
